src/lib/Persistence/Legacy/Bookmark/Gateway/DoctrineDatabase.php

```diff
@@ -100,6 +102,27 @@ public function loadBookmarkDataByUserIdAndLocationId(int $userId, array $locati
         return $query->execute()->fetchAll(PDO::FETCH_ASSOC);
     }
 
+    public function loadUserIdsByLocation(Location $location): array
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $queryBuilder
+            ->select(self::COLUMN_USER_ID)
+            ->from(self::TABLE_BOOKMARKS)
+            ->andWhere(
+                $queryBuilder
+                    ->expr()
+                    ->eq(
+                        self::COLUMN_LOCATION_ID,
+                        $queryBuilder->createNamedParameter(
+                            $location->id,
+                            Types::INTEGER
+                        )
+                    )
+            );
+
+        return $queryBuilder->execute()->fetchAllAssociative();
```

Comment:
PHPStan issue reported here can be mitigated by telling PHPStan the exact returned shape:
```suggestion

        /** @phpstan-var array<array{user_id: int}> */
        return $queryBuilder->execute()->fetchAllAssociative();
```

Reply:
And if you're only interested in first column, you can ask it for the first column only. There is a method on `Result` that allows that.

Reply:
https://github.com/ibexa/core/pull/417/commits/5d1b75e1d12cb3ef6fc803ef821d512b09e6384c